src/main/java/com/thebois/models/beings/Pawn.java

```diff
@@ -48,4 +59,63 @@ private void setRandomDestination() {
         setPath(newPath);
     }
 
+    protected void updateClosestStructure() {
+        if (closestStructure.isPresent() && getFinalDestination().isPresent()) {
+            // If the structure is completed or destination is unreachable pick a new one
+            if (closestStructure.get().isCompleted() || closestStructure.get().getPosition().equals(
+                getFinalDestination().get())) {
```

Comment:
question

According to the comment,
```java
closestStructure.get().getPosition().equals(getFinalDestination().get())
```
being true means that "destination is unreachable". I don't see how the position of the closest structure being equal to the final destination of the pawn implies that this is the case. How come?